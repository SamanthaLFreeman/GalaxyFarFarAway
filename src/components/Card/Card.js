import React from 'react';
import './Card.scss';

const Card = () => {
  return (
    <article className='card'>
      <h2 className='name'>Card</h2>
      <p className='homeworld'></p>
      <p className='species'></p>
      <p className='pop'></p>
      <button className='fav-btn'>Fav</button>
    </article>
  )
}

export default Card;