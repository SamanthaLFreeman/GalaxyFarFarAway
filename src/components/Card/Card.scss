$star-wars-font-color: #feda4a;
$main-font-color: aliceblue;

%fav-btn-shared {
    background: url('../../Images/star3.png') no-repeat center center;
    background-size: 50%;
    border: hidden;
    color: white;
    font-weight: bold;
    height: 60px;
    width: 60px;
}

.fav-btn:hover {
    background-size: 90%;
}

.fav-btn-active {
    @extend %fav-btn-shared;
    background-size: 90%;
}

.fav-btn {
    @extend %fav-btn-shared;
}

.cards {
    background: #2d445b;
    box-shadow:
        0 0 8px #fff,
        -1px 0 8px rgb(255, 140, 0), 
        1px 0 8px rgb(0, 30, 255);
    border-radius: 10%;
    color: $main-font-color;
    display: inline-flex;
    margin: 2%;
    padding: 15px;
    padding-top: 0;
    height: 315px;
    width: 25%;
}


.cards:hover  {
    box-shadow:
        0 0 15px #fff,
        -3px 0 18px rgb(255, 140, 0), 
        3px 0 18px rgb(0, 30, 255);
}

.fav-card {
    box-shadow:
        0 0 10px #fff,
        -2px 0 12px hsl(134, 100%, 50%), 
        2px 0 12px hsl(134, 100%, 50%);
}

.fav-card:hover {
    box-shadow:
        0 0 15px #fff,
        -3px 0 18px hsl(134, 100%, 50%), 
        3px 0 18px hsl(134, 100%, 50%);
}

h2.name {
    font-size: 1.5em;
    margin: 0;
}
