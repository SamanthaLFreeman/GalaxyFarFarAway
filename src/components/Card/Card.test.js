import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  it('should match the snapshot with all the data passed in correctly', () => {
    const wrapper = shallow(<Card />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke toggleFavorite method when the favorite button is clicked', () => {
    const toggleFavorite = jest.fn();
    const wrapper = shallow(<Card name={'Chewy'} type={'People'} toggleFavorite={toggleFavorite}/>)
    wrapper.find('button').simulate('click');

    expect(toggleFavorite).toHaveBeenCalledWith('Chewy', 'People');
  })
});