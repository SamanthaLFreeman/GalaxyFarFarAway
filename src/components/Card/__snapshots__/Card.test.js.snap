// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all the data passed in correctly 1`] = `
<article
  className="card"
>
  <h2
    className="name"
  >
    Card
  </h2>
  <p
    className="homeworld"
  />
  <p
    className="species"
  />
  <p
    className="pop"
  />
  <button
    className="fav-btn"
  >
    Fav
  </button>
</article>
`;
